web: Update strings from "Alert Words" to "Watched Phrases".

Aditya8840 opened the pull request:
This PR migrates user-facing strings and relevant code references from "Alert Words" to "Watched Phrases".
Currently, it does not include documentation changes and the handling of the `is:alerted` filter.

---

**Issue Link**  
Fixes part of #28779.

---
**Visual Changes**

| Description        | Image                                               |
|--------------------|-----------------------------------------------------|
| Watched phrases    | ![Watched phrases](https://github.com/user-attachments/assets/13b3f79a-afc3-4a94-bd72-df184a0bb86d) |
| `is:watched`         | ![is:watched](https://github.com/user-attachments/assets/0a0e7fea-5770-44d0-96db-9ada6f36606f) |

---

<details>
<summary>Self-review Checklist</summary>
<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>


---

Aditya8840 committed:
web: Update strings from "Alert Words" to "Watched Phrases".

This commit migrates user-facing strings and relevant code references
from "Alert Words" to "Watched Phrases".

Fixes part of #28779.

Aditya8840 commented on web/src/settings_config.ts:
```diff
@@ -839,7 +839,7 @@ export const all_notifications = (settings_object: Settings): AllNotifications =
         },
         {
-            label: $t({defaultMessage: "DMs, mentions, and alerts"}),
+            label: $t({defaultMessage: "DMs, mentions, and watched notifications"}),
```
Currently renamed "alerts" to "watched notifications". Need suggestions on naming this.

timabbott replied:
"DMs and mentions" is probably the best we can do for now. I think it's currently not the case that they can send email/push notifications anyway (#28779, etc.), so not mentioning those is probably an improvement.

You should do a clean commit series for this -- maybe a single prep commit that changes this string (`git grep 'DMs, mentions'`) across the app, including the Help Center, and then a next commit that's the "watched phrases" transition.

timabbott commented on web/templates/settings_overlay.hbs:
```diff
@@ -36,7 +36,7 @@
                     <li class="sidebar-item" tabindex="0" data-section="alert-words">
                         <i class="sidebar-item-icon fa fa-book" aria-hidden="true"></i>
-                        <div class="text">{{t "Alert words" }}</div>
+                        <div class="text">{{t "Watched phrases" }}</div>
```
Why is this capitalized in the screenshot?

Aditya8840 replied:
Ahh, I forgot to update the screenshot. I’ve now updated the screenshot.

timabbott commented on web/tests/search_suggestion.test.js:
```diff
@@ -394,7 +394,7 @@ test("empty_query_suggestions", () => {
     assert.equal(describe("is:starred"), "Starred messages");
     assert.equal(describe("is:mentioned"), "@-mentions");
-    assert.equal(describe("is:alerted"), "Alerted messages");
+    assert.equal(describe("is:alerted"), "Watched messages");
```
This probably needs to be "Messages with watched phrases"

timabbott commented:
Can you also add a commit adding the `is:watched` alias for the search operator? I think we'll want to merge that as part of the same PR as the string changes.

Aditya8840 committed:
web: Prepare migration of Alert words to Watched Phrases.

This commit updates references from 'DMs, mentions, and alerts' to
'DMs and mentions' as part of the ongoing transition from Alert Words
to Watched Phrases.

Aditya8840 commented on web/tests/search_suggestion.test.js:
```diff
@@ -516,7 +514,7 @@ test("check_is_suggestions", ({override, mock_template}) => {
     assert.equal(describe("-is:starred"), "Exclude starred messages");
     assert.equal(describe("-is:mentioned"), "Exclude @-mentions");
-    assert.equal(describe("-is:alerted"), "Exclude alerted messages");
+    assert.equal(describe("-is:watched"), "Exclude messages with watched phrases");
```
Currently renamed "Exclude alerted messages" to "Exclude messages with watched phrases". Need suggestions on naming this.

Aditya8840 commented:
Just a quick reminder, @prakhar1144 — this PR is ready for another round of review.

Aditya8840 commented:
Just a quick reminder — this PR is ready for another round of review.

timabbott commented on web/src/search_suggestion.ts:
```diff
@@ -689,7 +689,7 @@ function get_is_filter_suggestions(last: NarrowTerm, terms: NarrowTerm[]): Sugge
         {
             search_string: "is:alerted",
-            description_html: "alerted messages",
+            description_html: "messages with watched phrases",
```
Do we want "messages containing watched phrases" here?

timabbott commented on web/src/message_edit.ts:
```diff
@@ -1135,7 +1135,7 @@ export function save_message_row_edit($row: JQuery): void {
             historical: message.historical,
             collapsed: message.collapsed,
-            alerted: message.alerted,
+            watched: message.watched,
```
What sets `message.watched`?

timabbott commented:
The last commit looks busted; at least https://github.com/zulip/zulip/pull/32245#pullrequestreview-2480193144 seems like it's changing something beyond adding the `is:watched` alias, and instead expecting a change in how message objects are stored in the `message_store` structures to have a new `.watched` field. That should be in its own commit at the very least, but I'm also not sure if it's just broken (trying to read a value that isn't present).

timabbott commented:
@Aditya8840 can you clean up this PR and then ask @laurynmm to review?

Aditya8840 commented:
@laurynmm, Ready for your review!

Aditya8840 committed:
narrow: Add backend support for `is:watched` narrow.

This commit adds backend support for the `is:watched` narrow, while
retaining `is:alerted` for backward compatibility.

Aditya8840 committed:
search: Rename the search alias is:alerted to is:watched.

This commit renames the search filter "is:alerted" to
"is:watched".

Fixes part of zulip#28779.

Aditya8840 commented:
Changes:
- Added a third commit that introduces backend support for `is:watched`, with backward compatibility.

Aditya8840 commented:
@laurynmm, Ready for your review!

Aditya8840 commented on help/edit-a-message.md:
```diff
@@ -63,7 +63,7 @@ removed or changed from a regular mention to a
 If you [delete the content of a message](/help/delete-a-message#delete-message-content),
 any pending email notifications for that message will be canceled, including
-[mentions and alerts](/help/dm-mention-alert-notifications).
+[mentions](/help/dm-mention-alert-notifications).
```
I'm working on it!

Aditya8840 replied:
Currently, [alert words cannot trigger email or push notifications](https://github.com/zulip/zulip/pull/32245#discussion_r1830257915). So, do we still need to include that in the documentation?

laurynmm commented on help/dm-mention-alert-notifications.md:
```diff
@@ -54,25 +54,25 @@ channels in your [Channel settings](/help/channel-notifications), and
-{settings_tab|alert-words}
+{settings_tab|watched-phrases}
 
 1. Add a word or phrase.
```
This is out-of-date with the current UI, which has a button that opens a modal ...

laurynmm commented on web/templates/settings/alert_word_settings.hbs:
```diff
@@ -1,18 +1,18 @@
     <form id="alert_word_info_box">
         <p class="alert-word-settings-note">
-            {{t "Alert words allow you to be notified as if you were @-mentioned when certain words or phrases are used in Zulip. Alert words are not case sensitive."}}
+            {{t "Watched phrases allow you to be notified as if you were @-mentioned when certain words or phrases are used in Zulip. Watched phrases are not case sensitive."}}
```
This could also be improved in that prep commit ...

`Alert words allow you to receive desktop notifications as if you were @-mentioned when ...`

laurynmm commented on web/templates/settings_overlay.hbs:
```diff
@@ -35,7 +35,7 @@
                     {{/unless}}
                     <li class="sidebar-item" tabindex="0" data-section="alert-words">
                         <i class="sidebar-item-icon fa fa-book" aria-hidden="true"></i>
-                        <div class="text">{{t "Alert words" }}</div>
```
I think we'll want to change the URL hash for this tab of the personal settings overlay with these updates as well. See #26110 for when the "Display settings" tab was changed to "Preferences".

laurynmm commented on api_docs/unmerged.d/ZF-657917.md:
```diff
@@ -0,0 +1,8 @@
+  **Deprecated** `is:alerted` [search/narrow filter](/api/construct-narrow) in favor
+  of `is:watched`. The `is:alerted` filter continues to work for backwards compatibility,
+  but clients should use `is:watched` for consistency with the updated terminology
+  for watched phrases (previously called "alert words").
```
Let's revise this to match what's in the changelog entry for feature level 177.

laurynmm commented on zerver/tests/test_message_fetch.py:
```diff
@@ -955,6 +963,13 @@ def test_build_narrow_predicate(self) -> None:
+        self.assertFalse(
+            narrow_predicate(
+                message={},
+                flags=["watched"],
```
I don't think this is correct as we're not adding a watched flag ...

laurynmm commented on web/tests/filter.test.cjs:
```diff
@@ -357,6 +357,12 @@ test("basics", () => {
+    terms = [{operator: "is", operand: "alerted"}];
+    filter = new Filter(terms);
+    assert.ok(filter.has_operand("is", "watched"));
+    assert.ok(!filter.has_operand("is", "alerted"));
```
I'd also add a test for putting in "watched" as the operand here.

laurynmm commented on web/tests/search_suggestion.test.cjs:
```diff
@@ -166,7 +166,6 @@ test("dm_suggestions", ({override, mock_template}) => {
     suggestions = get_suggestions(query);
     expected = [
         "is:dm al",
-        "is:dm is:alerted",
```
I wonder if we want to do something for folks that have "alerted" in their muscle memory for search results. So that if they start typing 'al', there's a suggestion for 'is:watched'. We did something like this for the change from "pm-with" to "dm" operator change, I think.

alya replied:
Yeah, good thought. We might want to show `is:watched` as an auto-complete result for `is:alerted` prefixes.

alya replied:
Probably not a lot of downside, as there's unlikely to be a different reason to type `is:al`.

laurynmm reviewed (commented):
@Aditya8840 - Thanks for making those updates and apologies for taking a couple of weeks to get another review of these changes in.

I haven't yet pulled the PR down for QA testing, I've just gone through another read of the code and documentation changes. Let me know if you have any questions about my comments (either here or on CZO)!
